catalog/resource_sql_table.go

```diff
@@ -0,0 +1,383 @@
+
+	isView := ti.TableType == "VIEW"
+
+	externalFragment := ""
+	if ti.TableType == "EXTERNAL" {
+		externalFragment = "EXTERNAL "
+	}
+
+	createType := ti.getTableTypeString()
+
+	statements = append(statements, fmt.Sprintf("CREATE %s%s %s", externalFragment, createType, ti.FullName()))
+
+	if len(ti.ColumnInfos) > 0 {
+		statements = append(statements, fmt.Sprintf(" (%s)", ti.serializeColumnInfos()))
+	}
+
+	if !isView {
+		if ti.DataSourceFormat != "" {
+			statements = append(statements, fmt.Sprintf("\nUSING %s", ti.DataSourceFormat)) // USING CSV
+		}
+	}
+
+	if ti.Comment != "" {
+		statements = append(statements, fmt.Sprintf("\nCOMMENT '%s'", ti.Comment)) // COMMENT 'this is a comment'
+	}
+
+	if len(ti.Properties) > 0 {
+		statements = append(statements, fmt.Sprintf("\nTBLPROPERTIES (%s)", ti.serializeProperties())) // TBLPROPERTIES ('foo'='bar')
+	}
+
+	if !isView {
+		if ti.StorageLocation != "" {
+			statements = append(statements, "\n"+ti.buildLocationStatement())
+		}
+	} else {
+		statements = append(statements, fmt.Sprintf("\nAS %s", ti.ViewDefinition))
+	}
+
+	statements = append(statements, ";")
+
+	return strings.Join(statements, "")
+}
+
+func (ti *SqlTableInfo) updateTable(ctx context.Context, d *schema.ResourceData, s map[string]*schema.Schema, c *common.DatabricksClient, oldti *SqlTableInfo) error {
+	typestring := ti.getTableTypeString()
+
+	if ti.TableType == "VIEW" {
+		// View only attributes
+		if ti.ViewDefinition != oldti.ViewDefinition {
+			err := ti.applySql(fmt.Sprintf("ALTER VIEW %s AS %s", ti.FullName(), ti.ViewDefinition))
+			if err != nil {
+				return err
+			}
+		}
+	} else {
+		// Table only attributes
+		if ti.StorageLocation != oldti.StorageLocation {
+			err := ti.applySql(fmt.Sprintf("ALTER TABLE %s SET %s", ti.FullName(), ti.buildLocationStatement()))
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	// Attributes common to both views and tables
+	if ti.Comment != oldti.Comment {
+		err := ti.applySql(fmt.Sprintf("COMMENT ON %s %s IS '%s'", typestring, ti.FullName(), ti.Comment))
+		if err != nil {
+			return err
+		}
+	}
+
+	if !reflect.DeepEqual(ti.Properties, oldti.Properties) {
+		// First handle removal of properties
+		removeProps := make([]string, 0)
+		for key := range oldti.Properties {
+			if _, ok := ti.Properties[key]; !ok {
+				removeProps = append(removeProps, key)
+			}
+		}
+		if len(removeProps) > 0 {
+			err := ti.applySql(fmt.Sprintf("ALTER %s %s UNSET TBLPROPERTIES IF EXISTS (%s)", typestring, ti.FullName(), strings.Join(removeProps, ",")))
+			if err != nil {
+				return err
+			}
+		}
+		// Next handle property changes and additions
+		err := ti.applySql(fmt.Sprintf("ALTER %s %s SET TBLPROPERTIES (%s)", typestring, ti.FullName(), ti.serializeProperties()))
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (ti *SqlTableInfo) createTable(ctx context.Context, d *schema.ResourceData, s map[string]*schema.Schema, c *common.DatabricksClient) error {
```

Comment:
can you remove unused arguments, like `ctx`, `d`, `s`, and `c`?

Reply:
done